packages/calypso-build/webpack/sass.js

```diff
@@ -18,6 +18,29 @@ module.exports.loader = ( { includePaths, prelude, postCssOptions } ) => ( {
 			loader: require.resolve( 'css-loader' ),
 			options: {
 				importLoaders: 2,
+				modules: {
+					auto: /\.module\.scss$/, // Only enable CSS modules for .module.scss files
+					getLocalIdent: ( context, localIdentName, localName ) => {
+						// Only apply custom naming for .module.scss files
+						if ( ! /\.module\.scss$/.test( context.resourcePath ) ) {
+							// For non-modules, return the original class name (no hashing)
+							return localName;
+						}
+
+						// Custom class name for modules
+						return (
+							context.resourcePath.split( '/' ).slice( -2 ).join( '_' ) +
```

Comment:
Do we have any guarantees that this will produce [safe CSS names](https://developer.mozilla.org/en-US/docs/Web/CSS/ident)?

Reply:
`resourcePath` are valid file names, so I'm guessing so.

However, I think we can go with the simpler solution following the hashing in this file:

```js
require( 'crypto' )
    .createHash( 'md5' )
    .update( context.resourcePath + localName )
    .digest( 'hex' )
```

This seems to be enough.

Reply:
There are some weird gotchas like not allowing to start with a number which could happen with a file name or a hash value.

Looks like the default behavior has some extra checks to handle this:

https://github.com/webpack-contrib/css-loader/blob/9d030151bb221888ecfb0163b10910a7a918fe31/src/utils.js#L349-L356